examples/CMakeLists.txt

```diff
@@ -97,6 +89,10 @@ function(thrust_add_example target_name_var example_name example_src thrust_targ
   thrust_clone_target_properties(${example_target} ${thrust_target})
   thrust_fix_clang_nvcc_build_for(${example_target})
 
+  if ("CUDA" STREQUAL "${config_device}")
```

Comment:
I prefer examples using public CMake functions only so users can see how to use a project. So I would drop the usage of internal functions.